<a href="https://colab.research.google.com/github/KharchenkoAnastasia/customer_analysis_grocery_chain/blob/main/customer_analysis_grocery_chain.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Analysis of customers of a chain of grocery stores

In [15]:
import pandas as pd

In [16]:
path_to_file = '/content/Grocery Database.xlsx'
xl = pd.ExcelFile(path_to_file)
xl.sheet_names

['Grosto DB', 'Fact Summary']

In [17]:
xl.parse('Fact Summary')

S.No.          Column Name  \
0       1       Receipt Number   
1       2                 Date   
2       3                 Year   
3       4                Month   
4       5                 Time   
5       6        Mobile Number   
6       7        Membership ID   
7       8  Loyalty Card Points   
8       9                  Age   
9      10               Gender   
10     11                 City   
11     12              Country   
12     13             Category   
13     14         Sub_Category   
14     15                Items   
15     16               Brand    
16     17         Description    
17     18                Price   
18     19                  QTY   
19     20                 DISC   
20     21      Net Bill Amount   
21     22                  GST   
22     23    Gross Bill Amount   
23     24         Payment Mode   
24     25            Bank Name   
25     26      % Profit Margin   
26     27     % Operating Cost   
27     28       % Product Cost   
28     29        Profit Margin   
29     30       Operating Cost   
30     31         Product Cost   

                      Description of the Column names  
0                                  Transaction Number  
1                                 Date of Transaction  
2                                 Year of Transaction  
3                                Month of Transaction  
4                             Time of the transaction  
5                      Monbile number of the customer  
6                Unique Membership ID of the Customer  
7   Loyalty Points earned by customer (10 Points/1...  
8         Age of the Customer in the Company Database  
9                              Gender of the customer  
10                                 City Name of store  
11                              Country Name of store  
12                Category of Item bought by customer  
13            Sub-Category of Item bought by customer  
14               Name if the item  bought by customer  
15                           Brand of the Item bought  
16  Description of the Item (weight/volume/pack si...  
17                         Price per unit of the Item  
18                    Quantity bought by the cistomer  
19                          Discount offered by store  
20        Amount exclusive of tax payable to customer  
21                                        Tax Payable  
22  Gross Bill to be paid by customer inclusive of...  
23                  Mode of payment opted by customer  
24                                   Name of the Bank  
25                  Percentage profit margin of store  
26                 Percentage operating cost of store  
27                   Percentage product cost to store  
28                      Profit Margin earned by store  
29                 Operating cost beared by the store  
30                         Product Cost for the store

In [28]:
df = pd.read_excel(path_to_file)
df.head(10)

In [19]:
print(f"The dataset contains {df.shape[0]} rows and {df.shape[1]} columns.")

The dataset contains 50447 rows and 32 columns.


In [24]:
# Grouping the data by 'Membership ID' to check the distribution of customer memberships
df.groupby("Membership ID").size()

Membership ID
MIDID0001    38
MIDID0002    30
MIDID0003    31
MIDID0004    50
MIDID0005    27
             ..
MIDSG1278     3
MIDSG1279     2
MIDSG1280     4
MIDSG1281     7
MIDSG1282     7
Length: 2601, dtype: int64

In [25]:
# Creating a cross-tabulation to analyze the distribution of 'Membership ID' across different years
pd.crosstab(df['Year'], df['Membership ID'])

Membership ID  MIDID0001  MIDID0002  MIDID0003  MIDID0004  MIDID0005  \
Year                                                                   
2016                   6          1          6         10          4   
2017                   8          3          0         26         11   
2018                   9          5         12          0          2   
2019                  15         21         13         14         10   

Membership ID  MIDID0006  MIDID0007  MIDID0008  MIDID0009  MIDID0010  ...  \
Year                                                                  ...   
2016                   1          6         12          2         11  ...   
2017                  22          1          9          7          1  ...   
2018                  20          0         21         12          6  ...   
2019                   9          7          9          2          7  ...   

Membership ID  MIDSG1273  MIDSG1274  MIDSG1275  MIDSG1276  MIDSG1277  \
Year                                                                   
2016                   0          0          0          0          0   
2017                   6          0          0          0          0   
2018                   9          2          9          1          2   
2019                   0          7          0         10          1   

Membership ID  MIDSG1278  MIDSG1279  MIDSG1280  MIDSG1281  MIDSG1282  
Year                                                                  
2016                   0          0          0          0          0  
2017                   0          0          0          0          0  
2018                   0          0          0          0          0  
2019                   3          2          4          7          7  

[4 rows x 2601 columns]

  

> In total, the data frame contains data on 2,601 unique customers for the period 2016-2019. Accordingly, there are old and new customers who have different shopping experiences in the studied chain of stores.



In [27]:
# Displaying a concise summary of the dataset, including column data types, non-null counts, and memory usage
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50447 entries, 0 to 50446
Data columns (total 32 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Receipt Number       50447 non-null  object        
 1   Date                 50447 non-null  datetime64[ns]
 2   Year                 50447 non-null  int64         
 3   Month                50447 non-null  int64         
 4   Time                 50447 non-null  float64       
 5   Mobile Number        50447 non-null  int64         
 6   Membership ID        50447 non-null  object        
 7   Loyalty Card Points  50447 non-null  int64         
 8   Age                  50447 non-null  int64         
 9   Gender               50447 non-null  object        
 10  City                 50447 non-null  object        
 11  Country              50447 non-null  object        
 12  Category             50447 non-null  object        
 13  Sub_Category         50447 non-



> There are no missing values ​​in the dataframe.

